airbyte-integrations/connectors/destination-sftp-json/destination_sftp_json/client.py

```diff
@@ -0,0 +1,121 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+import contextlib
+import errno
+import json
+from pathlib import Path
+from typing import TextIO, Optional, Dict, Iterator, List
+
+import paramiko
+import smart_open
+
+
+@contextlib.contextmanager
+def sftp_client(
+    host: str,
+    port: int,
+    username: str,
+    password: str,
+) -> Iterator[paramiko.SFTPClient]:
```

Comment:
is this the desirable return type? if we're returning a single client why would it be iterable?

Reply:
That's a fair point! I think previously PyCharm would complain about the typing on `contextlib.contextmanager` return types, but that may have been resolved.